ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/sparse_encoding/SparseEncodingTranslator.java

```diff
@@ -0,0 +1,101 @@
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensors;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.*;
+
+import static org.opensearch.ml.common.CommonValue.ML_MAP_RESPONSE_KEY;
+
+public class SparseEncodingTranslator implements ServingTranslator {
+    private HuggingFaceTokenizer tokenizer;
+
+    @Override
+    public Batchifier getBatchifier() {
+        return Batchifier.STACK;
+    }
+    @Override
+    public void prepare(TranslatorContext ctx) throws IOException {
+        Path path = ctx.getModel().getModelPath();
+        tokenizer = HuggingFaceTokenizer.builder().optPadding(true).optTokenizerPath(path.resolve("tokenizer.json")).build();
+    }
+
+    @Override
+    public NDList processInput(TranslatorContext ctx, Input input) {
+        String sentence = input.getAsString(0);
+        NDManager manager = ctx.getNDManager();
+        NDList ndList = new NDList();
+        Encoding encodings = tokenizer.encode(sentence);
+        long[] indices = encodings.getIds();
+        long[] attentionMask = encodings.getAttentionMask();
+
+        NDArray indicesArray = manager.create(indices);
+        indicesArray.setName("input1.input_ids");
+
+        NDArray attentionMaskArray = manager.create(attentionMask);
+        attentionMaskArray.setName("input1.attention_mask");
+
+        ndList.add(indicesArray);
+        ndList.add(attentionMaskArray);
+        return ndList;
+    }
+    private Map<String, Float>  convertOutput(NDArray array)
+    {
+        Map<String, Float> map = new HashMap<>();
+        NDArray nonZeroIndices = array.nonzero().squeeze();
+
+        for (long index : nonZeroIndices.toLongArray()) {
+            String s = this.tokenizer.decode(new long[]{index}, true);
+            if (!s.isEmpty()){
+                map.put(s, array.getFloat(index));
+            }
+        }
+        return map;
+    }
+    @Override
+    public Output processOutput(TranslatorContext ctx, NDList list) {
+        Output output = new Output(200, "OK");
+
+        List<ModelTensor> outputs = new ArrayList<>();
+        Iterator<NDArray> iterator = list.iterator();
+        while (iterator.hasNext()) {
+            NDArray ndArray = iterator.next();
+            String name = ndArray.getName();
+            Map<String, Float> tokenWeightsMap = convertOutput(ndArray);
+            Map<String, ?> wrappedMap = Map.of(ML_MAP_RESPONSE_KEY, Collections.singletonList(tokenWeightsMap));
+            ModelTensor tensor = ModelTensor.builder()
+                    .name(name)
+                    .dataAsMap(wrappedMap)
+                    .build();
+            outputs.add(tensor);
+        }
+
+        ModelTensors modelTensorOutput = new ModelTensors(outputs);
+        output.add(modelTensorOutput.toBytes());
+        return output;
+    }
+
+    @Override
+    public void setArguments(Map<String, ?> arguments) {
```

Comment:
Do we need this empty method?

Reply:
Yes. This is the a interface must be override. Like SentenceTransformerTextEmbeddingTranslator:: setArguments